src/main.rs

```diff
@@ -19,6 +19,9 @@ struct Cli {
   #[arg(short, long, global = true)]
   pub verbose: bool,
 
+  #[arg(long, global = true, default_value = "hvm", help = "Path to hvm binary")]
+  pub hvm_path: String,
```

Comment:
I think you misunderstood what I said in my previous review, instead you fixed something that I missed, so it works well in the end.

What I meant is that hvm_path should go into RunArgs and GenArgs, rather than being a general argument (since check and gen-hvm don't depend on hvm). I got the names confused.

But on second thought this might actually be better to allow quickly changing modes using the same overall command, so I'll go with this.

Reply:
Great!